pkg/agent/apiserver/apiserver.go

```diff
@@ -155,6 +157,14 @@ func newConfig(bindPort int, enableMetrics bool, kubeconfig string) (*genericapi
 		GitCommit:    antreaversion.GetGitSHA(),
 	}
 	serverConfig.EnableMetrics = enableMetrics
+	// Add readiness probe to check the status of watchers.
+	check := healthz.NamedCheck("watcher", func(_ *http.Request) error {
+		if npq.GetControllerConnectionStatus() {
+			return nil
+		}
+		return fmt.Errorf("some watchers may not be connected")
+	})
```

Comment:
when I test the probe, I get `[-]watcher failed: reason withheld`

`reason withheld` is not what I'd expect here, shouldn't it say `some watchers may not be connected`?

Reply:
Seems withheld intentionally:
https://github.com/kubernetes/kubernetes/blob/480b4c74b858a9caac0b33cb4d9a1ab30ae65cf4/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go#L227-L231